### ISSUE TITLE
uploading with :context and progress bar display causes java.lang.ArithmeticException: Divide by zero

### ISSUE THREAD

**Issue by retrogradeorbit** [epiccastle/spire#29]
```
(upload {:content (selmer "provisioning/nginx-static.conf"
                          {:domain "mysite.mydomain"
                           :root "/var/www/mysite.mydomain/public"
                           :auth-basic {:name "access"
                                        :password-file "/var/www/mysite.mydomain/passwords"}})
         :dest "/root/test.conf"})
```

gives:

```
provisioning/production.clj:26 (upload {:content (selmer "provisioning/nginx-static.conf" {:root "/var/www/mysite.my
Exception in thread "main" clojure.lang.ExceptionInfo: Divide by zero [at line 6, column 1] {:type :sci/error, :line 6, :column 1, :message "Divide by zero [at line 6, column 1]"}
	at sci.impl.utils$rethrow_with_location_of_node.invokeStatic(utils.cljc:72)
	at sci.impl.utils$rethrow_with_location_of_node.invoke(utils.cljc:56)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:479)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:459)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:509)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_form.invokeStatic(interpreter.cljc:549)
	at sci.impl.interpreter$eval_form.invoke(interpreter.cljc:540)
	at sci.impl.interpreter$eval_string_STAR_.invokeStatic(interpreter.cljc:559)
	at sci.impl.interpreter$eval_string_STAR_.invoke(interpreter.cljc:552)
	at sci.impl.interpreter$eval_string$fn__2858.invoke(interpreter.cljc:572)
	at sci.impl.interpreter$eval_string.invokeStatic(interpreter.cljc:569)
	at sci.impl.interpreter$eval_string.invoke(interpreter.cljc:565)
	at sci.core$eval_string.invokeStatic(core.cljc:174)
	at sci.core$eval_string.invoke(core.cljc:137)
	at spire.core$evaluate.invokeStatic(core.clj:57)
	at spire.core$evaluate.invoke(core.clj:55)
	at spire.core$_main.invokeStatic(core.clj:96)
	at spire.core$_main.doInvoke(core.clj:73)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at spire.core.main(Unknown Source)
Caused by: java.lang.ArithmeticException: Divide by zero
	at clojure.lang.Numbers.divide(Numbers.java:188)
	at spire.utils$progress_stats.invokeStatic(utils.clj:185)
	at spire.utils$progress_stats.invoke(utils.clj:144)
	at spire.module.upload$upload_STAR_$progress_fn__5785.invoke(upload.clj:158)
	at spire.scp$scp_copy_data.invokeStatic(scp.clj:139)
	at spire.scp$scp_copy_data.invoke(scp.clj:108)
	at spire.scp$scp_to.invokeStatic(scp.clj:241)
	at spire.scp$scp_to.doInvoke(scp.clj:211)
	at clojure.lang.RestFn.invoke(RestFn.java:1843)
	at spire.module.upload$upload_STAR_.invokeStatic(upload.clj:226)
	at spire.module.upload$upload_STAR_.doInvoke(upload.clj:100)
	at clojure.lang.RestFn.invoke(RestFn.java:457)
	at sci.impl.interpreter$fn_call.invokeStatic(interpreter.cljc:425)
	at sci.impl.interpreter$fn_call.invoke(interpreter.cljc:425)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:475)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:459)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:509)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_do_STAR_$fn__2624.invoke(interpreter.cljc:382)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:382)
	at sci.impl.interpreter$eval_do_STAR_.invoke(interpreter.cljc:379)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:392)
	at sci.impl.interpreter$eval_do.invoke(interpreter.cljc:389)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:429)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:466)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:459)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:509)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_let$fn__2473.invoke(interpreter.cljc:74)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:62)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:59)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:433)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:466)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:459)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:509)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_do_STAR_$fn__2624.invoke(interpreter.cljc:382)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:382)
	at sci.impl.interpreter$eval_do_STAR_.invoke(interpreter.cljc:379)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:392)
	at sci.impl.interpreter$eval_do.invoke(interpreter.cljc:389)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:429)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:466)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:459)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:509)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_try.invokeStatic(interpreter.cljc:249)
	at sci.impl.interpreter$eval_try.invoke(interpreter.cljc:245)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:510)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_do_STAR_$fn__2624.invoke(interpreter.cljc:382)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:382)
	at sci.impl.interpreter$eval_do_STAR_.invoke(interpreter.cljc:379)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:392)
	at sci.impl.interpreter$eval_do.invoke(interpreter.cljc:389)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:429)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:466)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:459)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:509)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_do_STAR_$fn__2624.invoke(interpreter.cljc:382)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:382)
	at sci.impl.interpreter$eval_do_STAR_.invoke(interpreter.cljc:379)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:392)
	at sci.impl.interpreter$eval_do.invoke(interpreter.cljc:389)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:429)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:466)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:459)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:509)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_do_STAR_$fn__2624.invoke(interpreter.cljc:382)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:382)
	at sci.impl.interpreter$eval_do_STAR_.invoke(interpreter.cljc:379)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:392)
	at sci.impl.interpreter$eval_do.invoke(interpreter.cljc:389)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:429)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:466)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:459)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:509)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_try.invokeStatic(interpreter.cljc:249)
	at sci.impl.interpreter$eval_try.invoke(interpreter.cljc:245)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:510)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:83)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:59)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:433)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:466)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:459)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:509)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:83)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:59)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:433)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:466)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:459)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:509)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_do_STAR_$fn__2624.invoke(interpreter.cljc:382)
	at sci.impl.interpreter$eval_do_STAR_.invokeStatic(interpreter.cljc:382)
	at sci.impl.interpreter$eval_do_STAR_.invoke(interpreter.cljc:379)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:392)
	at sci.impl.interpreter$eval_do.invoke(interpreter.cljc:389)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:429)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:466)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:459)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:509)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_try.invokeStatic(interpreter.cljc:249)
	at sci.impl.interpreter$eval_try.invoke(interpreter.cljc:245)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:510)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:492)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:83)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:59)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at sci.impl.interpreter$eval_special_call.invokeStatic(interpreter.cljc:433)
	at sci.impl.interpreter$eval_special_call.invoke(interpreter.cljc:427)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:466)
	... 18 more
```

**Comment by hamann** [comment]
I'm frequently running into this. Is there a chance to get a new release in short because building the native image from source fails for some reason.

**Comment by retrogradeorbit** [comment]
Yes. Will release a new version now. Thanks for the patch aswell.

**Comment by retrogradeorbit** [comment]
fixed in 0.1.0-alpha.4
